### PR TITLE
Implement log forwarding with fluent-bit

### PR DESCRIPTION
**Pull request by dippynark**
Fixes: https://github.com/jetstack/tarmak/issues/182

TODOs:

- [x] Allow specifying of a logstashPrefix
- [x] Test behaviour with seperate audit/platform/application log
- [x] Ensure nothing gets deployed with log sinks disabled
- [x] Support HTTP basic auth for ES output
- [x] Support custom CA for ES output

```release-note
Add support for log forwarding with fluent-bit
```

### CODE REVIEW

File: puppet/modules/aws_es_proxy/templates/aws-es-proxy.service.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Documentation=https://github.com/abutaha/aws-es-proxy
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=<%= scope['proxy_path'] %> \
+  -listen localhost:<%= scope['listen_port'] %> \
+  -endpoint <%= scope['endpoint'] %> \
+  -verbose \
+  -pretty
```
Comment (simonswine) [inline puppet/modules/aws_es_proxy/templates/aws-es-proxy.service.erb]: please remove verbose and pretty

File: puppet/modules/fluent_bit/metadata.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "jetstack-kubernetes",
```
[the remark this answers is not in the document]
Comment (dippynark) [reply puppet/modules/fluent_bit/metadata.json]: The whole file or just that line?

File: puppet/modules/fluent_bit/labels.yaml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- name: size/XXL
+  color: ee0000
+- name: stale
+  color: "795548"
+- name: status/in-progress
+  color: fef2c0
+- name: status/in-review
+  color: fef2c0
+- name: triaged
+  color: d455d0
```
[the remark this answers is not in the document]
Comment (dippynark) [reply puppet/modules/fluent_bit/labels.yaml]: The whole file?

File: puppet/modules/fluent_bit/README.md [hunk 4]
```diff
@@ -0,0 +1,652 @@
+# kubernetes
```
[the remark this answers is not in the document]
Comment (dippynark) [reply puppet/modules/fluent_bit/README.md]: Which generator?

**Pull request merged**